instrumentation/opentelemetry-instrumentation-api/javaagent/build.gradle.kts

```diff
@@ -2,10 +2,27 @@ plugins {
   id("otel.javaagent-instrumentation")
 }
 
+// note that muzzle is not run against the current SNAPSHOT instrumentation-api, but this is ok
+// because the tests are run against the current SNAPSHOT instrumentation-api which will catch any
+// muzzle issues in SNAPSHOT instrumentation-api
+
+muzzle {
+  pass {
+    group.set("io.opentelemetry.instrumentation")
+    module.set("opentelemetry-instrumentation-api")
+    // currently all 1.0.0+ versions are alpha so they are all skipped
+    // if you want to test them anyways, comment out "alpha" from the exclusions in AcceptableVersions.kt
+    versions.set("[1.14.0-alpha,)")
+    assertInverse.set(true)
+    excludeInstrumentationModule("io.opentelemetry.javaagent.instrumentation.opentelemetryapi.OpenTelemetryApiInstrumentationModule")
```

Comment:
this is my solution (for now at least) to #798

Reply:
it's not pretty but...

Reply:
Instead of using class name, WDYT about using the instrumentation name? Most of them (or all; should be all) use the same name as gradle submodules, they're a lot shorter than the class names, and I suppose maybe consistent with how you disable instrumenations in runtime.